src/inet/UDPEndPointImplLwIP.cpp

```diff
@@ -100,7 +107,7 @@ CHIP_ERROR UDPEndPointImplLwIP::BindImpl(IPAddressType addressType, const IPAddr
 
     if (res == CHIP_NO_ERROR)
     {
-        res = chip::System::MapErrorLwIP(udp_bind(mUDP, &ipAddr, port));
+        res = chip::System::MapErrorLwIP(RunOnTCPIP([this, &ipAddr, port]() { return udp_bind(mUDP, &ipAddr, port); }));
```

Comment:
The scope of what gets run under the lock was much larger before.  Why is the scope narrowing ok?  This applies in a bunch of places.

Reply:
this is the right way. acquiring the lock or hopping on the TCPIP thread is only needed for the duration of the LwIP API call.

Reply:
I think that this could use a more thorough analysis than "this is the right way"....  Especially given the specific problem in GetPCB I pointed out below.  Someone needs to carefully check that none of these other functions have similar problems and put the reasons why they don't in the commit message.

Reply:
sure, i responded to this comment. in short, "core locking" is for the benefit of lwip, not our endpoint. if we need to protect endpoint's internal state, we should have a separate lock.